### DISCUSSION TITLE
Change component localisation/language

### DISCUSSION THREAD

**Discussion by joeDH00** [Budibase/budibase#14934] [Help]
Hi, I have a question regarding changing the language of the budibase component specifically the attachment field.

![image](https://github.com/user-attachments/assets/d5271e2d-c0b3-4bb5-9e48-4bdf0e75fc9b)

I was wondering if I can change this component language to use other language such as Malay. So instead of "Drag and drop your file. Select a file to upload from your computer", I can change it to use other language

**Comment by vienhoanglong** [comment]
Hi @joeDH00 

This is how I change the content of the attachment field component. I hope it helps you.
![Screenshot 2024-11-01 120054](https://github.com/user-attachments/assets/c123c5b8-8549-4d8d-b935-b1d635ece5a8)
Embed in JavaScript:
```javascript
parent.document.querySelector('.spectrum-IllustratedMessage-heading').textContent = 'Upload your file';
parent.document.querySelector('.spectrum-IllustratedMessage--cta .spectrum-IllustratedMessage-description label').textContent = '📷';
parent.document.querySelector('.spectrum-IllustratedMessage--cta .spectrum-IllustratedMessage-description').lastChild.textContent = 'Drag and drop files from your device';

**Reply by joeDH00** [reply]
I noticed it works in some form and in other cases just either nothing changed or return a javascript error: "Error while executing JS"

**Reply by vienhoanglong** [reply]
I fixed that like this, Embed text:

```html
<img style="display: none;" src onerror="
  setTimeout(() => {
    if (parent && parent.document) {
      parent.document.querySelector('.spectrum-IllustratedMessage-heading').textContent = 'Upload your file';
      parent.document.querySelector('.spectrum-IllustratedMessage--cta .spectrum-IllustratedMessage-description label').textContent = '📷';
      parent.document.querySelector('.spectrum-IllustratedMessage--cta .spectrum-IllustratedMessage-description').lastChild.textContent = 'Drag and drop files from your device';
    }
  }, 100);
">

**Reply by joeDH00** [reply]
Hi, sorry for the late response as I was preoccupied with urgent matters. I tried your suggestions and unfortunately still unable to fully resolved it. The first suggestion does work sometimes so I will continue to expand on that and try to find a solution

**Comment by mikesealey** [comment]
Hey @joeDH00 - this is a good suggestion and something we could certainly look into. It would be a fairly straightforward change to add a field to the settings to allow the builder to determine what gets displayed to the end user in a similar way to how the "No rows found" text is set in the table.

If you feel confident in tackling it, you could submit a PR, alternatively you could create a new issue and suggest it as an enhancement.

**Reply by joeDH00** [reply]
Hey, I would like to learn tackling and submit the PR but unfortunately have no idea where to start. I might just create a new issue and suggest it instead
